MekHQ/src/mekhq/service/MassRepairConfiguredOptions.java

```diff
@@ -0,0 +1,198 @@
+            int type = MassRepairOption.VALID_REPAIR_TYPES[i];
+
+            MassRepairSalvageDialog.MassRepairOptionControl mroc = dlg.getMassRepairOptionControlMap().get(type);
+
+            if (mroc == null) {
+                continue;
+            }
+
+            MassRepairOption mro = new MassRepairOption(type, mroc.getActiveBox().isSelected(),
+                    mroc.getMinSkillCBox().getSelectedIndex(), mroc.getMaxSkillCBox().getSelectedIndex(),
+                    (Integer) mroc.getMinBTHSpn().getValue(), (Integer) mroc.getMaxBTHSpn().getValue());
+
+            if (mro.isActive()) {
+                setHasActiveMassRepairOption(true);
+            }
+
+            getMassRepairOptions().add(mro);
+        }
+    }
+    //endregion Initialization
+
+    //region Getters/Setters
+    public boolean isUseExtraTime() {
+        return useExtraTime;
+    }
+
+    public void setUseExtraTime(boolean useExtraTime) {
+        this.useExtraTime = useExtraTime;
+    }
+
+    public boolean isUseRushJob() {
+        return useRushJob;
+    }
+
+    public void setUseRushJob(boolean useRushJob) {
+        this.useRushJob = useRushJob;
+    }
+
+    public boolean isAllowCarryover() {
+        return allowCarryover;
+    }
+
+    public void setAllowCarryover(boolean allowCarryover) {
+        this.allowCarryover = allowCarryover;
+    }
+
+    public boolean isOptimizeToCompleteToday() {
+        return optimizeToCompleteToday;
+    }
+
+    public void setOptimizeToCompleteToday(boolean optimizeToCompleteToday) {
+        this.optimizeToCompleteToday = optimizeToCompleteToday;
+    }
+
+    public boolean isUseAssignedTechsFirst() {
+        return useAssignedTechsFirst;
+    }
+
+    public void setUseAssignedTechsFirst(boolean useAssignedTechsFirst) {
+        this.useAssignedTechsFirst = useAssignedTechsFirst;
+    }
+
+    public boolean isScrapImpossible() {
+        return scrapImpossible;
+    }
+
+    public void setScrapImpossible(boolean scrapImpossible) {
+        this.scrapImpossible = scrapImpossible;
+    }
+
+    public boolean isReplacePodParts() {
+        return replacePodParts;
+    }
+
+    public void setReplacePodParts(boolean replacePodParts) {
+        this.replacePodParts = replacePodParts;
+    }
+
+    public List<MassRepairOption> getMassRepairOptions() {
+        return massRepairOptions;
+    }
+
+    public void setMassRepairOptions(List<MassRepairOption> massRepairOptions) {
+        this.massRepairOptions = massRepairOptions;
+    }
+
+    public boolean hasActiveMassRepairOption() {
+        return hasActiveMassRepairOption;
+    }
+
+    public void setHasActiveMassRepairOption(boolean hasActiveMassRepairOption) {
+        this.hasActiveMassRepairOption = hasActiveMassRepairOption;
+    }
+    //endregion Getters/Setters
+
+    public List<MassRepairOption> getActiveMassRepairOptions() {
+        if (!hasActiveMassRepairOption()) {
+            return new ArrayList<>();
```

Comment:
If this should be immutable then prefer `Collections.emptyList()`. (realizing this was perhaps inherited code)